docs/decisions/0005-lti-provider-1.3-spec-support.rst

```diff
@@ -0,0 +1,685 @@
+   - Get related graded resources from data received. - Get all
+   assignments related - Increment the version value of each assignment
+   - Determine each assignment type and send the corresponding task
+   (composite module, or single problem). - Send the task to update the
+   score for each assignment by sending a message back to the platform
+   related to the resource. (Example:
+   https://github.com/openedx/edx-platform/pull/27411/files#diff-36022deef8607c7a4647c8f2620b4d9ed283d5b41077e966bfd097585e0ebe7cR480)
+
+Approach A: Create a new app 
+----------------------------
+
+This approach would be similar to the one proposed in the `PR
+21435 <https://github.com/openedx/edx-platform/pull/21435>`__). We will
+create a new app (Example: lti1p3_tool), and integrate all logic related
+to LTI 1.3 separated from the existing lti_provider app.
+
+Approach B: Refactor the lti_provider app
+-----------------------------------------
+
+We could refactor the existing lti_provider. With this approach, we
+could separate the logic of each specification per folder, and keep
+utilities that are used on both specifications on the base folder of the
+app, here is an example of the folder structure:
+
+.. code:: bash
+
+   .
+   ├── __init__.py
+   ├── apps.py
+   ├── admin.py
+   ├── tasks.py
+   ├── urls.py
+   ├── management
+   │   ├── commands
+   │   │   ├── __init__.py
+   │   │   └── ...
+   ├── migrations
+   │   ├── __init__.py
+   │   └── ...
+   ├── models.py
+   │   ├── __init__.py # LtiUser and GradedAssignment model could be  here
+   │   ├── 1p1 # Create LtiUser1p1 using multi-table inheritance with LtiUser
+   │   └── 1p3 # Create LtiUser1p3 using multi-table inheritance with LtiUser
+   ├── utils
+   │   ├── __init__.py
+   │   ├── users.py
+   │   ├── 1p3 # Store utilities for 1.3 specification
+   │   └── 1p1 # Store utilities for 1.1 specification
+   │       ├── __init__.py
+   │       ├── signature_validator.py
+   │       └── outcomes.py
+   ├── signals.py # Use score_changed_handler for both specifications
+   └── views.py
+       ├── __init__.py
+       ├── 1p1 # Define LTI 1.1 views here
+       └── 1p3 # Define LTI 1.3 views here
+
+Approach C: Create LTI Provider Plugin
+--------------------------------------
+
+Another approach would be to create a new openedx-plugin. We would
+create a new app that will include a common set of utilities and
+services used by both LTI 1.1 and LTI 1.3, and an app for each LTI
+specification, and their corresponding services, each one with their
+views and tasks. Similar to the structure of the B approach. This
+approach would also avoid adding more code to the monolithic
+edx-platform structure.
+
+In this approach, we will require to import a few classes and functions
+from edx-platform, for example, the User model to keep track of users on
+LTI launches the PROBLEM_WEIGHTED_SCORE_CHANGED signal to receive the
```

Comment:
The signal part can be done using an openedx-event. Now, the user model is a bit trickier. There are things we can do with the [configurable backends pattern](https://github.com/eduNEXT/eox-core/tree/master/eox_core/edxapp_wrapper/backends) to avoid the import, but we would still be tying the underlying implementation of the model to the plugin.